src/worker.cpp

```diff
@@ -1367,4 +1367,83 @@ void Worker::performBackupAndRestore(types::VPDMapVariant& io_srcVpdMap)
         */
     }
 }
+
+void Worker::deleteFruVpd(const std::string& i_dbusObjPath)
+{
+    try
+    {
+        const bool l_presentPropValue =
+            std::get<bool>(dbusUtility::readDbusProperty(
+                constants::pimServiceName, i_dbusObjPath,
+                constants::inventoryItemInf, "Present"));
+
+        if (!l_presentPropValue)
+        {
+            logging::logMessage(
+                "Given FRU is not present. Could not perform deletion for " +
+                i_dbusObjPath);
+            return;
+        }
+        else
+        {
+            std::vector<std::string> l_interfaceList{
+                constants::operationalStatusInf};
+            types::MapperGetSubTree l_subTreeMap =
+                dbusUtility::getObjectSubTree(i_dbusObjPath, 0,
+                                              l_interfaceList);
+
+            for (const auto& [l_objectPath, l_serviceInterfaceMap] :
+                 l_subTreeMap)
+            {
+                // ToDo -- Update functional property to true.
+                types::ObjectMap l_objectMap{
```

Comment:
Present property is being set to false here, again at the end "resetDataUnderPIM" is being called which will set the present property to false. 
Can you revisit the flow and check if these calls can be optimized. Also can call to PIM be done only once?

Reply:
Here we are checking for the sub FRU's and updating the present property for the sub FRU's